src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Common.cs

```diff
@@ -111,28 +113,36 @@ private static void SwapViews(FrameworkElement oldView, FrameworkElement newView
 			}
 #endif
 
+#if !HAS_UNO
 			if (oldView is FrameworkElement oldViewAsFE && newView is FrameworkElement newViewAsFE)
 			{
-				ApplyDataContext(parentDataContext, oldViewAsFE, newViewAsFE);
+				ApplyDataContext(parentDataContext, oldViewAsFE, newViewAsFE, oldDataContext);
 			}
+#endif
 		}
 
-		private static void ApplyDataContext(object? parentDataContext, FrameworkElement oldView, FrameworkElement newView)
+#if !HAS_UNO
+		private static void ApplyDataContext(
+			object? parentDataContext,
+			FrameworkElement oldView,
+			FrameworkElement newView,
+			object? oldDataContext)
 		{
 			if (oldView == null || newView == null)
 			{
 				return;
 			}
 
 			if ((newView.DataContext is null || newView.DataContext == parentDataContext)
-				&& (oldView.DataContext is not null && oldView.DataContext != parentDataContext))
+				&& (oldDataContext is not null && oldDataContext != parentDataContext))
```

Comment:
@nickrandolph nice find!! Do you think it's possible to create a runtime test?

Reply:
The sequence of events for this particular change is android-specific, which can't test in CI, and we're already validating DataContext changes in current regression tests.